### DISCUSSION TITLE
Where's the API doc?

### DISCUSSION THREAD

**Discussion by clementhervy** [goharbor/harbor#20987] [Q&A]
I'm looking for the API-specific, detailed documentation and I can't seem to find it. Information like what do the various endpoints models describe, what do the fields do, what values can they take...

To be more precise, I'm trying to create retention policies programmatically. The [documentation for that](https://goharbor.io/docs/2.0.0/working-with-projects/working-with-images/create-tag-retention-rules/) is entirely webapp-focused. I can't seem to find simple examples, like "if you want to keep images with this & that tag, send this model to this endpoint" which means that I'm either bad at googling or that basic documentation is inaccessible to search engine crawlers, somehow.

**Comment by MinerYang** [comment]
You could refer to our devcenter for more details
https://demo.goharbor.io/devcenter-api-2.0

**Comment by clementhervy** [comment]
Thank you for your answer. Of course I had seen the Swagger UI, but it does not in itself tells you everything you need to know about how to structure requests, which fields are required, what validation is done behind the scenes, etc.
To learn all that you have to check out the requests in the browser dev tools and read the code, which I mean, sure, it's there, yeah. It tells you what you need to know in the end, but all in all it makes using the API a poor experience.

**Reply by Vad1mo** [reply]
yeas the docs aren't as polished, as a tip, I can recommend to look into the interaction between the UI and the API, to understand better the parameters and responses. We are happy to accept contributions and examples in the API spec.

**Comment by MinerYang** [comment]
Thanks for your feedback. Unfortunately, we do not have any api specific doc yet only the swagger UI itself. We would like to hence this part for more guidance of using API.
